### ISSUE TITLE
The network takes a lot more disk space than expected

### ISSUE THREAD

**Issue by LordAddy** [fchollet/keras#6374]
Hello, I am confused when it comes to the size of the network saved on the hdd. It takes up `2.4 GB` on disk, when I would expect it to take only `810 MB`.

When I initialize the model and save it to disk before learning, it really takes up `810 MB` but once I train it, it takes up `2.4 GB` what are the two additional parameters stored per weight and is it possible to save the model in the basic representation?

I am using Theano backend. Also, this network barely fits into my 6 GB VRAM, I have to use batch size of 48 so that it doesn't overflow. How many variables are there per weight? I am using Adam optimizer, so I would expect 4 variables: the weight/bias, the first momentum, the second momentum, the cumulative gradient. Then a single parameter per neuron for storing the current gradient w.r.t. its inputs but that's only a few. So that still adds up to `4 * 810 = 3240 MB` but in reality, it takes `1.3 GB` more VRAM. What is the overhead?

Thanks in advance!

**Comment by LordAddy** [comment]
Okay, so I have looked into the model save file and found out that the extra parameters are indeed the first and second momentum for each weight from the Adam optimizer. I presume that when one wants to save just the network for production, doing `model.optimizer = None` would be sufficient.

Concerning the VRAM memory, I am still not aware of any way how to predict how much space will the whole learning actually require. I guess it is a hard question to answer because different types of layers demand a different amount of scratch space for the intermediate calculations, so it probably depends also on the structure of the network not just on the number of parameters. Is that a correct assumption?

**Comment by yhenon** [comment]
Use `model.save_weights` instead of `model.save` to save just the weights.
